### PR TITLE
Manual DNS Auth and Deploy Hook

### PR DESCRIPTION
**Pull request by ikogan**
This adds support for a deploy hook to all custom deployment of certificates as well as support for Certbot's manual authorization method with hooks.

### CODE REVIEW

File: letsencrypt/client/init.sls [hunk 1]
```diff
@@ -102,9 +102,66 @@ certbot_cron:
...
+    - template: jinja
     - require:
       - cmd: certbot_installed
 
 {%- endif %}
 
+{%- if client.hook %}
+certbot_deploy_hook:
+    file.managed:
+        - name: /usr/local/lib/certbot/hooks/deploy-hook
```
Comment (fpytloun) [inline letsencrypt/client/init.sls]: Isn't better to use /etc/certbot as a location for these hooks instead of introducing this path?

**Comment by fpytloun** [comment]
Thank you for contribution :-) Please see my comment above.

File: letsencrypt/client/init.sls (hunk 1 above)
Comment (ikogan) [reply letsencrypt/client/init.sls]: You're probably right, I forgot to look into that. I don't have an `/etc/certbot`, is that created automatically by something? Is `/etc/certbot/hooks` the correct place for them?

Comment (fpytloun) [reply letsencrypt/client/init.sls]: Sorry, it's `/etc/letsencrypt/`
Didn't find any recommendation but that location probably makes more sense.

Comment (ikogan) [reply letsencrypt/client/init.sls]: There _is_ a `renewal-hooks` directory but I worry that some other script automatically runs hooks out of there under some circumstances. Do you know if that's the case? Is it safe to drop the deploy hook into `/etc/letsencrypt/renewal-hooks/deploy`?

Comment (fpytloun) [reply letsencrypt/client/init.sls]: Unfortunately I don't know :-/ Would prefer using some custom directory, eg. `scripts`

**Pull request closed without merging**